src/models/transaction.rs

```diff
@@ -77,12 +78,21 @@ pub fn rpc_to_ec_recovered_transaction(
     let signature = transaction.signature.ok_or(SignatureError::MissingSignature)?;
     let transaction = rpc_to_primitive_transaction(transaction)?;
 
+    let parity = match transaction.tx_type() {
+        TxType::Legacy => {
+            let chain_id = transaction.chain_id().ok_or(TransactionError::InvalidChainId)?;
+            let recovery: U256 = U256::from(2) * U256::from(chain_id) + U256::from(35);
+            signature.v - recovery
+        }
+        TxType::Eip1559 | TxType::Eip2930 | TxType::Eip4844 => signature.v,
+    };
```

Comment:
Can we put a comment here just to say:

// EIP-155: v = {0, 1} + CHAIN_ID * 2 + 35

In order to understand the root cause of this formula? wdyt?